app/src/main/java/org/schabi/newpipe/player/PopupVideoPlayer.java

```diff
@@ -1082,25 +1106,52 @@ public boolean onTouch(View v, MotionEvent event) {
         private boolean handleMultiDrag(final MotionEvent event) {
             if (event.getPointerCount() != 2) return false;
 
-            final float firstPointerX = event.getX(0);
-            final float secondPointerX = event.getX(1);
+            if(initPointerDistance != -1){
 
-            final float diff = Math.abs(firstPointerX - secondPointerX);
-            if (firstPointerX > secondPointerX) {
-                // second pointer is the anchor (the leftmost pointer)
-                popupLayoutParams.x = (int) (event.getRawX() - diff);
-            } else {
-                // first pointer is the anchor
-                popupLayoutParams.x = (int) event.getRawX();
-            }
+                //get the movements of the fingers
+                float firstPointerMoveX = event.getX(0) - initFirstPointerX;
+                float firstPointerMoveY = event.getY(0) - initFirstPointerY;
+                float secPointerMoveX = event.getX(1) - initSecPointerX;
+                float secPointerMoveY = event.getY(1) - initSecPointerY;
+                //minimum threshold beyond which pinch gesture will work
+                int scaledTouchSlop = ViewConfiguration.get(PopupVideoPlayer.this).getScaledTouchSlop();
 
-            checkPopupPositionBounds();
-            updateScreenSize();
+                if(firstPointerMoveX > scaledTouchSlop ||firstPointerMoveY > scaledTouchSlop ||
+                        secPointerMoveX > scaledTouchSlop || secPointerMoveY > scaledTouchSlop){
 
-            final int width = (int) Math.min(screenWidth, diff);
-            updatePopupSize(width, -1);
+                    double newWidth = popupWidth;
 
-            return true;
+                    //calculate current distance between the pointers
+                    float currentXDiff = event.getX(0) - event.getX(1);
+                    float currentYDiff = event.getY(0) - event.getY(1);
+                    double currentPointerDistance = Math.sqrt(currentXDiff * currentXDiff + currentYDiff * currentYDiff);
```

Comment:
There is the [Java Math.hypot() function](https://www.tutorialspoint.com/java/lang/math_hypot.htm) that does exactly this
```suggestion
                    double currentPointerDistance = Math.hypot(event.getX(0) - event.getX(1), event.getY(0) - event.getY(1));
```